libr/crypto/p/crypto_ed25519.c

```diff
@@ -0,0 +1,109 @@
+/* radare - LGPL - Copyright 2024 - Sylvain Pelissier
+ * Implementation of Ed25519 signature algorithm (RFC 8032)
+ * Based on Orson Peters implementation: https://github.com/orlp/ed25519 */
+
+#include <r_lib.h>
+#include <r_crypto.h>
+#include <r_util/r_log.h>
+#include "../signature/ed25519/ge.h"
+#include "../signature/ed25519/sc.h"
+#include "../hash/sha2.h"
+
+#define ED25519_SIG_LEN 64
+
+static bool ed25519_set_key(RCryptoJob *cj, const ut8 *key, int keylen, int mode, int direction) {
+	if (!(keylen == 32 || keylen == 64)) {
+		return false;
+	}
+	if (keylen == 32) {
+		// Using a seed
+		RHash *ctx = r_hash_new (true, R_HASH_SHA512);
```

Comment:
at some point during r2-5.9.9, we should use hash instances to avoid creating new instances of the hash to perform a checksum